tests/_models/test_dag.py

```diff
@@ -72,6 +72,11 @@ def test_package_dag_filter_nonexistent_packages(t_fnmatch: PackageDAG) -> None:
         t_fnmatch.filter_nodes(["x", "y", "z"], None)
 
 
+def test_package_dag_filter_nonexistent_packages_uses_pep503normalize(t_fnmatch: PackageDAG) -> None:
+    with pytest.raises(ValueError, match="No packages matched using the following patterns: X.Y, Y_Z, Z-X"):
+        t_fnmatch.filter_nodes(["X.Y", "Y_Z", "Z-X"], None)
+
+
```

Comment:
What I mean is I want to see if we normalize user input.

Like this:
```suggestion
def test_package_dag_filter_nonexistent_packages_uses_pep503normalize(
    mock_pkgs: Callable[[MockGraph], Iterator[Mock]],
) -> None:
    graph: MockGraph = {
        ("Pie.Pie", "1"): [],
    }
    pkgs = PackageDAG.from_pkgs(list(mock_pkgs(graph)))
    pkgs = pkgs.filter_nodes(["Pie.Pie"], None)
    assert len(pkgs) == 1
    assert pkgs.get_node_as_parent("pie-pie") is not None
```

Reply:
Got it, and I finished some others.

Reply:
Done again.